### PR TITLE
[PDI-17853] Need support to connect with MongoDB Atlas using latest S…

### PR DESCRIPTION
**Pull request by GopiKrishnaAngadi**
…RV connection string

### CODE REVIEW

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 1m 18s
Build command:
```console
mvn clean verify -B -e -Daudit -amd
```
---
:ok_hand: All tests passed!
 
Tests run: **80**, Failures: **0**, Skipped: **0** &nbsp;&nbsp; [Test Results](http://wingman.pentaho.net/job/projects/job/mongo-utils/job/PR-82/1/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

File: src/main/java/org/pentaho/mongo/wrapper/DefaultMongoClientFactory.java [hunk 1]
```diff
@@ -17,10 +17,7 @@
 
 package org.pentaho.mongo.wrapper;
 
-import com.mongodb.MongoClient;
-import com.mongodb.MongoClientOptions;
-import com.mongodb.MongoCredential;
-import com.mongodb.ServerAddress;
+import com.mongodb.*;
```
Comment (peterrinehart) [inline src/main/java/org/pentaho/mongo/wrapper/DefaultMongoClientFactory.java:20]: Change this section back; our style guidelines specify that each import must be included separately, even if we're importing the entire package.

File: src/main/java/org/pentaho/mongo/wrapper/ConnectionStringMongoClientWrapper.java [hunk 2]
```diff
@@ -0,0 +1,371 @@
+package org.pentaho.mongo.wrapper;
```
Comment (peterrinehart) [inline src/main/java/org/pentaho/mongo/wrapper/ConnectionStringMongoClientWrapper.java:1]: This file is missing the Hitachi license header.

File: src/test/java/org/pentaho/mongo/wrapper/ConnectionStringMongoClientWrapperTest.java [hunk 3]
```diff
@@ -0,0 +1,365 @@
+package org.pentaho.mongo.wrapper;
```
Comment (peterrinehart) [inline src/test/java/org/pentaho/mongo/wrapper/ConnectionStringMongoClientWrapperTest.java:1]: This file is missing the Hitachi license header.

**Review by peterrinehart — CHANGES_REQUESTED** [review]

File: src/main/java/org/pentaho/mongo/wrapper/DefaultMongoClientFactory.java (hunk 1 above)
Comment (GopiKrishnaAngadi) [reply src/main/java/org/pentaho/mongo/wrapper/DefaultMongoClientFactory.java:20]: Changed

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 1m 13s
Build command:
```console
mvn clean verify -B -e -Daudit -amd
```
---
:ok_hand: All tests passed!
 
Tests run: **80**, Failures: **0**, Skipped: **0** &nbsp;&nbsp; [Test Results](http://wingman.pentaho.net/job/projects/job/mongo-utils/job/PR-82/2/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 52s
Build command:
```console
mvn clean verify -B -e -Daudit -amd
```
---
:ok_hand: All tests passed!
 
Tests run: **80**, Failures: **0**, Skipped: **0** &nbsp;&nbsp; [Test Results](http://wingman.pentaho.net/job/projects/job/mongo-utils/job/PR-82/3/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 1m 34s
Build command:
```console
mvn clean verify -B -e -Daudit -amd
```
---
:ok_hand: All tests passed!
 
Tests run: **80**, Failures: **0**, Skipped: **0** &nbsp;&nbsp; [Test Results](http://wingman.pentaho.net/job/projects/job/mongo-utils/job/PR-82/4/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

File: src/main/java/org/pentaho/mongo/wrapper/ConnectionStringMongoClientWrapper.java [hunk 4]
```diff
@@ -0,0 +1,380 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package org.pentaho.mongo.wrapper;
+
+import com.mongodb.*;
```
Comment (peterrinehart) [inline src/main/java/org/pentaho/mongo/wrapper/ConnectionStringMongoClientWrapper.java:19]: This import needs to be changed; we have to import all classes from a package separately, even if we end up importing the entire package.

File: src/main/java/org/pentaho/mongo/wrapper/ConnectionStringMongoClientWrapper.java [hunk 5]
```diff
@@ -0,0 +1,380 @@
+/*!
+ * Copyright 2010 - 2021 Hitachi Vantara.  All rights reserved.
```
Comment (peterrinehart) [inline src/main/java/org/pentaho/mongo/wrapper/ConnectionStringMongoClientWrapper.java:2]: If this file is new as of this year, the only date it needs is 2021.  The dates on each file should reflect how long the file has been in the baseline.

**Review by peterrinehart — CHANGES_REQUESTED** [review]
Additional style changes needed.

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 1m 11s
Build command:
```console
mvn clean verify -B -e -Daudit -amd
```
---
:ok_hand: All tests passed!
 
Tests run: **80**, Failures: **0**, Skipped: **0** &nbsp;&nbsp; [Test Results](http://wingman.pentaho.net/job/projects/job/mongo-utils/job/PR-82/5/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Pull request merged**